main.cpp

```diff
@@ -56,46 +56,107 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    float value;
+    const char* name;
+    T(const float v, const char* ptr) :  //1
+    value(0.0f),//2
+    name(nullptr)//3
+    {
+        value = v;
+        if(ptr != nullptr)
+        {
+            name = ptr;
+        }
+        else std::cout << "name is nullptr!\n";
+    }
```

Comment:
It is perfectly fine to initialize `std::string` with a null `const char*`
remove all of this extra code in the constructor
initialize `name` with `ptr`
initialize `value` with `v` 
Use the member initializer list correctly
- [ ] revise

Reply:
I think I better understand the member initializer list now